### PR TITLE
Prune/jsonnet namespace roles

### PR DESCRIPTION
**Pull request by prune998**
This PR replace the Jsonnet config that create the RBAC roles for `default` and `kube-system` namespaces by a loop that will generate the RBAC rules for a list of namespaces.
The default list includes `default`, `kube-system` and the Prometheus install namespace (`monitoring` by default).

I also added a simple doc file.

### CODE REVIEW

**Comment by brancz** [comment]
You just need to run `make generate-in-docker` in the root of this repo, and commit the generation changes, and then this should be good to go!

File: contrib/kube-prometheus/docs/monitoring-other-namespaces.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Monitoring other Kubernetes Namespaces
+This guide will help you monitor applications in other Namespaces. By default the RBAC rules are only enabled for the `Default` and `kube-system` Namespace during Install.
+
+# Setup
+You have to give the list of the Namespaces that you want to be able to monitor.
+This is done in the variable `prometheus.roleSpecificNamespaces`. You usually set this in your `.jsonnet` file when building the manifests.
+
+Ex to create the needed `Role` and `Rolebindig` for the Namespace `foo` :
```
Comment (mxinden) [inline contrib/kube-prometheus/docs/monitoring-other-namespaces.md]: `s/Ex/Example/`

**Comment by prune998** [comment]
@mxinden it's done !

**Comment by brancz** [comment]
lgtm :+1: thanks a lot!

**Pull request merged**